File: testdata/bench/javascript/BenchmarkTest00019.js
// Vulnerable: SQL injection via header value in query
const express = require('express');
const db = require('./db');
const router = express.Router();
router.get('/test019', (req, res) => {
    const apiKey = req.headers['x-api-key'];
    const sql = "SELECT * FROM api_keys WHERE key_value = '" + apiKey + "'";
    db.query(sql, (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00019.rs
use actix_web::{web, HttpResponse};

#[derive(serde::Deserialize)]
struct SearchReq { query: String }

// Vulnerable: SQL injection from JSON body
async fn handler(body: web::Json<SearchReq>) -> HttpResponse {
    let param = body.query.clone();
    let sql = format!("SELECT * FROM documents WHERE content LIKE '%{}%'", param);
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
